src/CargoBuild.UnitTests/CargoBuildTest.cs

```diff
@@ -0,0 +1,424 @@
+
+            cargoBuildProject.TryBuild(restore: true, out bool result, out BuildOutput buildOutput);
+
+            result.ShouldBeTrue(buildOutput.GetConsoleLog());
+
+            buildOutput.Messages.High.ShouldContain("D031211C98F1454CA47A424ADC86A8F7", buildOutput.GetConsoleLog());
+        }
+
+        [Fact]
+        public void DoNotReferenceOutputAssemblies()
+        {
+            ProjectCreator projectA = ProjectCreator.Templates.SdkCsproj(
+                    path: Path.Combine(TestRootPath, "ProjectA", "ProjectA.csproj"),
+#if NETFRAMEWORK || NET8_0
+                    targetFramework: "net8.0")
+#elif NET9_0
+                    targetFramework: "net9.0")
+#endif
+                .Save();
+
+            ProjectCreator cargoBuildProject = ProjectCreator.Templates.CargoBuildProject(
+                    path: Path.Combine(TestRootPath, "CargoBuild", "Rust.cargoproj"))
+                .ItemProjectReference(projectA)
+                .Save();
+
+            cargoBuildProject.TryRestore(out bool result, out BuildOutput buildOutput);
+
+            result.ShouldBeTrue(buildOutput.GetConsoleLog());
+        }
+
+        [Theory]
+        [InlineData(".cargoproj")]
+        [InlineData(".csproj")]
+        [InlineData(".proj")]
+        public void ProjectContainsStaticGraphImplementation(string projectExtension)
+        {
+            ProjectCreator cargoBuild = ProjectCreator.Templates.CargoBuildProject(
+                path: GetTempFileWithExtension(projectExtension),
+                globalProperties: new Dictionary<string, string>
+                {
+                    ["IsGraphBuild"] = bool.TrueString,
+                },
+                customAction: creator =>
+                {
+                    creator.Target("TakeAction", afterTargets: "Build")
+                        .TaskMessage("86F00AF59170450E9D687652D74A6394", MessageImportance.High);
+                })
+                .Property("GenerateDependencyFile", "false")
+                .Save();
+
+            ICollection<ProjectItem> projectReferenceTargets = cargoBuild.Project.GetItems("ProjectReferenceTargets");
+
+            TargetProtocolShouldContainValuesForTarget("Build");
+            TargetProtocolShouldContainValuesForTarget("Clean");
+            TargetProtocolShouldContainValuesForTarget("Rebuild");
+            TargetProtocolShouldContainValuesForTarget("Publish");
+
+            void TargetProtocolShouldContainValuesForTarget(string target)
+            {
+                IEnumerable<string> buildTargets =
+                    projectReferenceTargets.Where(i => i.EvaluatedInclude.Equals(target, StringComparison.OrdinalIgnoreCase))
+                        .Select(i => i.GetMetadata("Targets")?.EvaluatedValue)
+                        .Where(t => !string.IsNullOrEmpty(t));
+
+                buildTargets.ShouldNotBeEmpty();
+            }
+        }
+
+        [Fact]
+        public void ProjectsCanDependOnCargoBuildProjects()
```

Comment:
Isn't this backwards? The cargo projects are depending on the csproj.

Also, I don't see anything in the SDK which indicates that a ProjectReference does anything but enforce a build ordering. I thought we had talked about, for example, a vcxproj referencing a rust project and being able to wire that up correctly? I'm not seeing anything in the SDK that would make that work properly.